web-bundle/src/main/java/com/graphhopper/resources/IsochroneResource.java

```diff
@@ -124,19 +119,71 @@ public Response doGet(
                 feature.setGeometry(geometryFactory.createPolygon(polygonShell));
                 features.add(feature);
             }
+            ObjectNode json = JsonNodeFactory.instance.objectNode();
+            json.putPOJO("polygons", features);
+            sw.stop();
+            logger.info("took: " + sw.getSeconds() + ", visited nodes:" + isochrone.getVisitedNodes() + ", " + uriInfo.getQueryParameters());
+            return Response.fromResponse(jsonSuccessResponse(json, sw.getSeconds()))
+                    .header("X-GH-Took", "" + sw.getSeconds() * 1000)
+                    .build();
+        } else if ("pointlist".equalsIgnoreCase(resultStr)) {
+            Collection<String> header = new LinkedHashSet<>(Arrays.asList("longitude", "latitude", "time", "distance"));
+            if (!Helper.isEmpty(extendedHeader))
+                header.addAll(Arrays.asList(extendedHeader.split(",")));
+            List<Isochrone.IsoLabelWithCoordinates> resultList = isochrone.search(qr.getClosestNode());
+            List<List> items = new ArrayList(resultList.size());
+            for (Isochrone.IsoLabelWithCoordinates label : resultList) {
+                List list = new ArrayList(header.size());
+                for (String h : header) {
+                    switch (h) {
+                        case "distance":
+                            list.add(label.distance);
+                            break;
+                        case "time":
+                            list.add(label.time);
+                            break;
+                        case "node_id":
+                            list.add(label.adjNodeId);
+                            break;
+                        case "edge_id":
+                            list.add(label.edgeId);
+                            break;
+                        case "longitude":
+                            list.add(label.adjCoordinate.x);
+                            break;
+                        case "latitude":
+                            list.add(label.adjCoordinate.y);
+                            break;
+                        case "prev_longitude":
```

Comment:
Can we please add options to export previous node's time and distance, so we can get values in both ends of segment without a lookup join?
```
                        case "prev_distance":
                            list.add(Optional.ofNullable(index.get(label.baseNodeId)).map(i -> i.distance).orElse(0D));
                            break;
                        case "prev_time":
                            list.add(Optional.ofNullable(index.get(label.baseNodeId)).map(i -> i.time).orElse(0L));
                            break;
```

Reply:
Do you have an example why this would be useful for you?

Reply:
Yeah. Builiding a constrained Delaunay triangulation and clipping it in PostGIS, I need to have two-point segments to feed into function. Here's more detail: https://www.patreon.com/posts/isochrones-are-20933638